staNMF/staNMF.py

```diff
@@ -61,27 +64,44 @@ class staNMF:
 
     :param: seed (int, optional with default 123): sets numpy random seed
 
-    :param: replicates(int, optional with default 100): Number of bootstrapped
+    :param: replicates(int or tuple of ints of length 2, optional with default
+    int 100):
+    Specifies which/ how many bootstrapped repetitions to be performed on each
+    value of K, for use in stability analysis; if a list of length 2 is given,
+    self.replicates is set to a list of ints between the first and second
+    elements of this tuple. If it is set to an integer, self.replicates is set
+    to a list of ints between 0 and the given integer
+
+    Number of bootstrapped
     repetitions of NMF to be performed on each value of K, for use in stability
     analysis
 
     :param: NMF_finished (bool, optional with default False): True if runNMF
     has been completed for the dataset and k range for which you wish to
     calculate instability. To surpass NMF step if fileID file already contains
     factorization solutions for X in your range [K1, K2], set to True
+
+    :param: parallel (bool, optional with default False): True if NMF is to be
+    run in parallel such that the instability calculation should write a file
+    for each K containing its instability index
+
     '''
 
     def __init__(self, filename, folderID="", K1=15, K2=30,
                  sample_weights=False, seed=123, replicates=100,
-                 NMF_finished=False):
+                 NMF_finished=False, parallel=False):
         warnings.filterwarnings("ignore")
         self.K1 = K1
         self.K2 = K2
         self.sample_weights = sample_weights
         self.seed = seed
         self.guess = np.array([])
         self.guessdict = {}
-        self.replicates = replicates
+        self.parallel = parallel
+        if isinstance(replicates, int):
+            self.replicates = range(replicates)
+        elif isinstance(replicates, tuple) and len(replicates):
```

Comment:
This is the same as `elif isinstance(replicates, tuple) and replicates:`.

I'd recommend:

```python
elif isinstance(replicates, tuple):
    start, stop = replicates
    self.replicates = range(replicates[0], replicates[1])
```

Will then throw error if replicates is not length 2.